server/rpc/handle_get_block_template.go

```diff
@@ -131,6 +129,17 @@ func handleGetBlockTemplate(s *Server, cmd interface{}, closeChan <-chan struct{
 	}
 }
 
+func isSyncedForMining(s *Server) bool {
```

Comment:
Please add an explanation of how the node chooses whether it's synced for mining.